toolchain/check/handle_impl.cpp

```diff
@@ -205,14 +205,17 @@ static auto PopImplIntroducerAndParamsAsNameComponent(
   auto [implicit_params_loc_id, implicit_param_patterns_id] =
       context.node_stack().PopWithNodeIdIf<Parse::NodeKind::ImplForall>();
 
-  ParameterBlocks parameter_blocks{
-      .implicit_params_id = SemIR::InstBlockId::Invalid,
-      .params_id = SemIR::InstBlockId::Invalid,
-      .return_slot_id = SemIR::InstId::Invalid};
   if (implicit_param_patterns_id) {
-    parameter_blocks =
+    auto parameter_blocks =
         CalleePatternMatch(context, *implicit_param_patterns_id,
                            SemIR::InstBlockId::Invalid, SemIR::InstId::Invalid);
+    CARBON_CHECK(parameter_blocks.calling_convention_params_id ==
+                     SemIR::InstBlockId::Empty,
+                 "{0} {1}", parameter_blocks.calling_convention_params_id,
+                 context.inst_blocks()
+                     .Get(parameter_blocks.calling_convention_params_id)
+                     .size());
```

Comment:
Would it be worth elaborating on this CHECK message with explanatory text? What's an example look like in a failure?

(or, if this was temporary for debugging an issue, should it be removed?)